### PR TITLE
fix(css-generator): remove TS4.1 requirements

### PR DESCRIPTION
**Pull request by peterblazejewicz**
The package latest version was created 3 years ago.
Switch to abstract class feature (the other could be exporting type/interface
and not using classes)

Thanks!

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

File: types/css-generator/index.d.ts [hunk 1]
```diff
@@ -2,15 +2,12 @@
...
-// TypeScript Version: 4.1
 
 export interface Options {
     indentation?: string;
 }
 
-export class Generator {
-    private constructor(options?: Options);
-
+export abstract class Generator {
```
Comment (septs) [inline types/css-generator/index.d.ts:10]: why?

**Comment by septs** [comment]
@peterblazejewicz this pr title is typo error?

File: types/css-generator/index.d.ts [hunk 2]
```diff
@@ -2,15 +2,12 @@
 // Project: https://github.com/luizbills/css-generator.js
 // Definitions by: Septs <https://github.com/septs>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 4.1
 
 export interface Options {
     indentation?: string;
 }
 
-export class Generator {
```
Comment (septs) [inline types/css-generator/index.d.ts:11]: plz revert, thanks

**Review by septs — CHANGES_REQUESTED** [review]

File: types/css-generator/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/css-generator/index.d.ts:10]: the private ctor class requires 4.1. This DT is used to consume a package or the purpose of the DT types
is to be used with package itself? (I cannot see reference in the package to use that types).
By cnverting to abstract we are removing the TS version dependency, so the DT can be used and composed with
types that are not necessarily top-notch. 
So the purpose is interoperability. thx!
But up to review, thx!

**Review by septs — APPROVED** [review]

**Comment by peterblazejewicz** [comment]
Ready to merge 💘
@septs thx!

**Pull request merged**